packages/integration-tests/src/tests/api/application/application-custom-data.test.ts

```diff
@@ -0,0 +1,24 @@
+import { ApplicationType } from '@logto/schemas';
+
+import {
+  createApplication,
+  getApplication,
+  patchApplicationCustomData,
+} from '#src/api/application.js';
+
+describe('application custom data API', () => {
+  it('should patch application custom data successfully', async () => {
+    const applicationName = 'test-create-app';
+    const applicationType = ApplicationType.SPA;
+    const application = await createApplication(applicationName, applicationType);
+
+    const customData = { key: 'value' };
+    const result = await patchApplicationCustomData(application.id, customData);
```

Comment:
can add a test case for reseting `customData` of some app per offline discussion.

Reply:
Added